src/wallet/wallet.h

```diff
@@ -431,11 +432,13 @@ class CWalletTx
     //! make sure balances are recalculated
     void MarkDirty()
     {
+        fChangeCached = false;
+        if (m_is_cache_empty) return;
```

Comment:
I don't think we should have an early return. The reset shouldn't be very expensive, and I think it's better to be sure that the cache is cleared when `m_is_cache_empty = true`

Reply:
Done.